File: algorithm/JavaScript/MinimizeMaximumPairSumInArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var minPairSum = function(nums) {
   nums.sort((a, b) => a - b)
   len = nums.length
   let index = 0, res = -Infinity
   while (index <= len / 2) {
     const temp = nums[index] + nums[len - index - 1]
     res = Math.max(res, temp)
     index += 1
   }
   return res

};

File: algorithm/python/MinimizeMaximumPairSumInArray.py
class Solution:
    def minPairSum(self, nums):
      nums.sort()
      res = float('-inf')
      length = len(nums)
      index = 0
      while index <= length // 2:
        temp = nums[index] + nums[length-index-1]
        res = max(res, temp)
        index += 1
      
      return res
